components/pipeline-service/development/increase-results-pruner-gracePeriod.yaml

```diff
@@ -16,6 +16,7 @@ spec:
               "token",
               "-check_owner=false",
               "-completed_run_grace_period",
-              # default pipeline-service configuration has a greacePeriod of "10m" and will be moving towards immediate pruning in future
+              # default pipeline-service configuration has a grace period of "10m"; for potential debug in CI
```

Comment:
While you're at it Gabe, could please replace this whole file with 
```yaml
- op: add
  path: /spec/template/spec/containers/1/args/-
  value: "-completed_run_grace_period=2h"
```
It should also fix the cpu & mem req change taking place in other container. Thank you

Reply:
yeah turns out the container ordering manifests differently with the developer overlay @ramessesii2 @psturc than the stage/prod overlays since the rbac proxy is added directly

that was not obvious to me when I first attempted this

I've fixed the "-completed_run_grace_period=2h" bit by editing the yaml file directly vs. the patch @ramessesii2 

to fix the container ordering, modifying the patch file to reference container 0 was what worked for me

I have verified locally via `kubectl kustomize -o <file>` on my local branch:

```
kind: Deployment
metadata:
  annotations:
    argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
    argocd.argoproj.io/sync-wave: "2"
  labels:
    app.kubernetes.io/name: tekton-results-watcher
    app.kubernetes.io/part-of: tekton-results
    app.kubernetes.io/version: devel
  name: tekton-results-watcher
  namespace: tekton-results
spec:
  replicas: 2
  selector:
    matchLabels:
      app.kubernetes.io/name: tekton-results-watcher
  template:
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
        cluster-autoscaler.kubernetes.io/safe-to-evict: "false"
      labels:
        app.kubernetes.io/name: tekton-results-watcher
        app.kubernetes.io/version: devel
    spec:
      affinity:
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
            - matchExpressions:
              - key: kubernetes.io/os
                operator: NotIn
                values:
                - windows
        podAntiAffinity:
          preferredDuringSchedulingIgnoredDuringExecution:
          - podAffinityTerm:
              labelSelector:
                matchLabels:
                  app.kubernetes.io/name: tekton-results-watcher
              topologyKey: kubernetes.io/hostname
            weight: 100
      containers:
      - args:
        - -api_addr
        - tekton-results-api-service.tekton-results.svc.cluster.local:8080
        - -auth_mode
        - token
        - -check_owner=false
        - -completed_run_grace_period=2h
        env:
        - name: SYSTEM_NAMESPACE
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace
        - name: CONFIG_LOGGING_NAME
          value: tekton-results-config-logging
        - name: CONFIG_LEADERELECTION_NAME
          value: tekton-results-config-leader-election
        - name: CONFIG_OBSERVABILITY_NAME
          value: tekton-results-config-observability
        - name: METRICS_DOMAIN
          value: tekton.dev/results
        - name: TEKTON_RESULTS_API_SERVICE
          value: tekton-results-api-service.tekton-pipelines.svc.cluster.local:8080
        - name: AUTH_MODE
          value: token
        image: quay.io/redhat-appstudio/tekton-results-watcher:2eef63e8a52cdc9529ea7ba0b0b8b0c19a8f160e
        name: watcher
        ports:
        - containerPort: 9090
          name: metrics
        - containerPort: 8008
          name: profiling
        resources:
          limits:
            cpu: 250m
            memory: 2Gi
          requests:
            cpu: 100m
            memory: 64Mi
```

Reply:
lgtm